src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -147,9 +151,36 @@ public override void SetLength(long value)
             throw new NotSupportedException(SR.NotSupported_UnseekableStream);
         }
 
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            CheckReadArguments(buffer, offset, count);
```

Comment:
Exceptions thrown from `CheckReadArguments` will be propagated on Task instead of throwing to user immediately.